opencensus-shim/src/test/java/io/opentelemetry/opencensusshim/OpenTelemetryMetricExporterTest.java

```diff
@@ -104,11 +104,16 @@ void testSupportedMetricsExportedCorrectly() throws Exception {
           .untilAsserted(
               () ->
                   assertThat(
+                          // Filter for metrics with name in allowedMetrics, and dedupe to only one
+                          // metric per unique metric name
```

Comment:
Was able to recreate the test flake by adding a `Thread.sleep(TimeUnit.SECONDS.toMillis(20))` above line 102. The test logic wasn't resilient to multiple metric exports taking place before the assertions are made.